### ISSUE TITLE
Writing checkpoints

### ISSUE THREAD

**Issue by tatiana-boura** [data-eng/navgreen#30]
Corresponding experiments and checks should be conducted in order to succesfully being able to write to the checkpoints' registers.

**Comment by stasinos** [comment]
Noting here that after designing the mechanism for remotely setting checkpoints, some of the code in PR #33 might need to re-visited, as it will be possible to simplify it under the assumption that checkpoints will not be set via other channels.

**Comment by tatiana-boura** [comment]
Since in  branch [write-to-PLC](https://github.com/data-eng/navgreen/tree/write-to-PLC) we started working on this Issue, let us state the problem with detail and provide some constraints set from the domain expert @ge0rgemera.

We have decided to work with a simpler case, at least for now, that involves the ground source (BTES) solar assisted heat pump. In this case, we can control the hp by changing the value of the corresponding **setpoint**. The setpoint has to have a value in range $[15, 51]$, but it cannot be changed arbitrarily. Specifically:

- If the hp *general alarm* is on, no operation on the hp is acceptible.
- The hp should not be turned on/off in a time period less than *5 minutes*.
- If the top layer temperature exceedes *52 oC*, the hp must be turned off.
- If the temperature of the ground BTES tank is less than *8 oC*, the hp should be turned off. If this case is true, no other operation on the hp setpoint is acceptible until the temperature of the BTES tank becomes $\geq 12\ oC$, in which senario the hp must be turned back on.
